packages/create-plugin/src/migrations/utils.ts

```diff
@@ -132,3 +133,174 @@ export function installNPMDependencies(context: Context) {
     execSync(installCmd, { cwd: context.basePath, stdio: 'inherit' });
   }
 }
+
+export function readJsonFile<T extends object = any>(context: Context, path: string): T {
+  if (!context.doesFileExist(path)) {
+    throw new Error(`Cannot find ${path}`);
+  }
+  try {
+    return JSON.parse(context.getFile(path) || '{}');
+  } catch (e) {
+    throw new Error(`Cannot parse ${path}: ${e}`);
+  }
+}
+
+export function addDependenciesToPackageJson(
+  context: Context,
+  dependencies: Record<string, string>,
+  devDependencies: Record<string, string> = {},
+  packageJsonPath = 'package.json'
+) {
+  const currentPackageJson = readJsonFile(context, packageJsonPath);
+  const currentDeps = { ...(currentPackageJson.dependencies || {}) };
+  const currentDevDeps = { ...(currentPackageJson.devDependencies || {}) };
+
+  // Handle dependencies
+  for (const [dep, newVersion] of Object.entries(dependencies)) {
+    if (currentDeps[dep]) {
+      if (isIncomingVersionGreater(newVersion, currentDeps[dep])) {
+        currentDeps[dep] = newVersion;
+      }
+    } else if (currentDevDeps[dep]) {
+      // Exists in devDependencies, only update there if new version is greater
+      if (isIncomingVersionGreater(newVersion, currentDevDeps[dep])) {
+        currentDevDeps[dep] = newVersion;
+      }
+    } else {
+      // Not present, add to dependencies
+      currentDeps[dep] = newVersion;
+    }
+  }
```

Comment:
Nit: do we want to log something if the migration was trying to downgrade dependencies?

Reply:
I've added some debug logging to the function. I don't think we want to output this for every migration, it could get noisy quickly.